src/main/java/in/projecteka/consentmanager/dataflow/DataFlowBroadcastListener.java

```diff
@@ -0,0 +1,79 @@
+import reactor.core.publisher.Mono;
+
+import javax.annotation.PostConstruct;
+
+import static in.projecteka.consentmanager.ConsentManagerConfiguration.HIP_DATA_FLOW_REQUEST_QUEUE;
+import static in.projecteka.consentmanager.clients.ClientError.queueNotFound;
+
+@AllArgsConstructor
+public class DataFlowBroadcastListener {
+    private static final Logger logger = Logger.getLogger(ConsentArtefactBroadcastListener.class);
+    private MessageListenerContainerFactory messageListenerContainerFactory;
+    private DestinationsConfig destinationsConfig;
+    private Jackson2JsonMessageConverter converter;
+    private DataFlowNotifier dataFlowNotifier;
+    private DataFlowRequestRepository dataFlowRequestRepository;
+    private ClientRegistryClient clientRegistryClient;
+
+    @PostConstruct
+    public void subscribe() throws ClientError{
+        DestinationsConfig.DestinationInfo destinationInfo = destinationsConfig
+                .getQueues()
+                .get(HIP_DATA_FLOW_REQUEST_QUEUE);
+        if (destinationInfo == null) {
+            logger.error(HIP_DATA_FLOW_REQUEST_QUEUE + " not found");
+            throw queueNotFound();
+        }
+
+        MessageListenerContainer mlc = messageListenerContainerFactory
+                .createMessageListenerContainer(destinationInfo.getRoutingKey());
+
+        MessageListener messageListener = message -> {
+            DataFlowRequestMessage dataFlowRequestMessage =
+                    (DataFlowRequestMessage) converter.fromMessage(message);
+            logger.info("Received message for Request id : " + dataFlowRequestMessage
+                    .getTransactionId());
+            DataFlowRequest dataFlowRequest = DataFlowRequest.builder()
+                    .transactionId(dataFlowRequestMessage.getTransactionId())
+                    .callBackUrl(dataFlowRequestMessage.getDataFlowRequest().getCallBackUrl())
+                    .consent(dataFlowRequestMessage.getDataFlowRequest().getConsent())
+                    .hiDataRange(dataFlowRequestMessage.getDataFlowRequest().getHiDataRange())
+                    .build();
+
+            dataFlowRequestRepository.getHipIdFor(dataFlowRequest.getConsent().getId())
+                    .flatMap(hipId -> providerUrl(hipId))
+                    .flatMap(url -> dataFlowNotifier.notifyHip(dataFlowRequest, url))
+                    .block();
+
+        };
+        mlc.setupMessageListener(messageListener);
+
+        mlc.start();
+    }
+
+    private Mono<String> providerUrl(String providerId) {
+        return clientRegistryClient.providerWith(providerId)
+                .flatMap(provider -> provider.getIdentifiers()
+                        .stream()
+                        .filter(Identifier::isOfficial)
+                        .findFirst()
+                        .map(identifier -> Mono.just(identifier.getSystem()))
+                        .orElse(Mono.empty()));
```

Comment:
Log the error if we can't find the URL of HIP, and find a way to put the request into dead-letter-queue.